scraper/src/fcc2zim/entrypoint.py

```diff
@@ -0,0 +1,219 @@
+        type=str,
+        help="Curriculum language",
+        required=True,
+    )
+    parser.add_argument(
+        "--name",
+        type=str,
+        help="ZIM name. Used as identifier and filename (date will be appended)",
+        required=True,
+    )
+    parser.add_argument(
+        "--title",
+        type=str,
+        help="Title of zim file",
+        required=True,
+    )
+    parser.add_argument(
+        "--description", type=str, help="Description of ZIM file", required=True
+    )
+    parser.add_argument(
+        "--long-description",
+        type=str,
+        help="Long description of ZIM file",
+    )
+    parser.add_argument(
+        "--creator",
+        type=str,
+        help="Name of freeCodeCamp courses creator",
+        default="freeCodeCamp",
+    )
+    parser.add_argument(
+        "--publisher", type=str, help="Publisher of the zim file", default="OpenZIM"
+    )
+    parser.add_argument(
+        "--fetch",
+        help="Run fetch scraper phase",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "--prebuild",
+        help="Run pre-build scraper phase",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "--build",
+        help="Run build scraper phase",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "--force",
+        help="Force a full reprocessing, not benefiting from any cached file",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "--debug",
+        help="Enable verbose output",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "--out-dir",
+        type=str,
+        help="Output directory where zim file will be built",
+        default="/output",
+    )
+    parser.add_argument(
+        "--tmp-dir",
+        type=str,
+        help="The temporary directory to hold temporary files during scraper operation",
+        default="/tmp",  # noqa: S108
+    )
+    parser.add_argument(
+        "--zimui-dist-dir",
+        type=str,
+        help=(
+            "Directory containing Vite build output from the Zim UI Vue.JS application"
+        ),
+        default="/src/zimui",
+    )
+    parser.add_argument(
+        "--zim-file",
+        type=str,
+        help="ZIM file name (based on --name if not provided), could contain {period}"
+        " placeholder which will be replaced by <year>_<month>",
+    )
+    parser.add_argument(
+        "--zip-path",
+        help="Path to zip file containing FCC courses",
+        type=str,
+    )
+    parser.add_argument(
+        "--version",
+        help="Display scraper version and exit",
+        action="version",
+        version=f"fcc2zim {VERSION}",
+    )
+
+    args = parser.parse_args()
+
+    Global.logger.info(f"Starting scraper {VERSION}")
+
+    set_debug(debug=args.debug)
+
+    do_fetch = args.fetch
```

Comment:
I feel like being able to run only one phase is very useful when debugging.
I like to have it rather than having to comment out code.

Reply:
You find it useful because this scraper is still an _old-fashion_ one that does a lot of things in the filesystem. Ultimately, this will be greatly reduced and processing will be done on the fly while creating the ZIM file.
That will render all this pretty insignificant.
We've had this discussion in the past, with @kelson42 and I know where this is going. We can keep it until you see the light 😉
In the mean time though, please remove those debug-only options from the cli and use environ to toggle those switches.

Reply:
I've probably already mostly seen the light (even if I'm not sure to totally have understood your position regarding the use of the filesystem with a SQLite DB, which is also a pain from my experience).